### DISCUSSION TITLE
bitmap交并差计算查询导致be节点崩溃

### DISCUSSION THREAD

**Discussion by dangerousfeng** [apache/doris#21612] [A - General / Q&A]
在进行bitmap类型字段交并差计算查询，be节点挂掉，be.out报以下信息：

*** Aborted at 1688709332 (unix time) try "date -d @1688709332" if you are using GNU date ***
*** Current BE git commitID: afe6bb9 ***
*** SIGSEGV address not mapped to object (@0x7effd4800000) received by PID 29537 (TID 29920 OR 0x7f038c132700) from PID 18446744072979742720; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/src/doris/be/src/common/signal_handler.h:413
 1# os::Linux::chained_handler(int, siginfo*, void*) in /data/app/jdk1.8.0_361/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /data/app/jdk1.8.0_361/jre/lib/amd64/server/libjvm.so
 3# signalHandler(int, siginfo*, void*) in /data/app/jdk1.8.0_361/jre/lib/amd64/server/libjvm.so
 4# 0x00007F044D525400 in /lib64/libc.so.6
 5# memcpy at /root/src/doris/be/src/glibc-compatibility/memcpy/memcpy_x86_64.cpp:219
 6# array_container_write in /data/app/doris-be-2.0.0-beta/lib/doris_be
 7# ra_portable_serialize in /data/app/doris-be-2.0.0-beta/lib/doris_be
 8# doris::detail::Roaring64Map::write(char*) const at /root/src/doris/be/src/util/bitmap_value.h:698
 9# doris::vectorized::DataTypeBitMap::serialize(doris::vectorized::IColumn const&, char*, int) const at /root/src/doris/be/src/vec/data_types/data_type_bitmap.cpp:68
10# doris::vectorized::Block::serialize(int, doris::PBlock*, unsigned long*, unsigned long*, doris::segment_v2::CompressionTypePB, bool) const at /root/src/doris/be/src/vec/core/block.cpp:816
11# doris::vectorized::VDataStreamSender::serialize_block(doris::vectorized::Block*, doris::PBlock*, int) at /root/src/doris/be/src/vec/sink/vdata_stream_sender.cpp:708
12# doris::vectorized::VDataStreamSender::send(doris::RuntimeState*, doris::vectorized::Block*, bool) at /root/src/doris/be/src/vec/sink/vdata_stream_sender.cpp:570
13# doris::PlanFragmentExecutor::open_vectorized_internal() in /data/app/doris-be-2.0.0-beta/lib/doris_be
14# doris::PlanFragmentExecutor::open() at /root/src/doris/be/src/runtime/plan_fragment_executor.cpp:273
15# doris::FragmentExecState::execute() at /root/src/doris/be/src/runtime/fragment_mgr.cpp:263
16# doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::RuntimeState*, doris::Status*)> const&) at /root/src/doris/be/src/runtime/fragment_mgr.cpp:527
17# std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)> const&)::$_0>::_M_invoke(std::_Any_data const&) at /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
18# doris::ThreadPool::dispatch_thread() in /data/app/doris-be-2.0.0-beta/lib/doris_be
19# doris::Thread::supervise_thread(void*) at /root/src/doris/be/src/util/thread.cpp:466
20# start_thread in /lib64/libpthread.so.0
21# clone in /lib64/libc.so.6

**Comment by yagagagaga** [comment]
Can you provide the sql which can let me reproduce?

**Reply by dangerousfeng** [reply]
不方便贴全部语句，但是目前已找到解决方式，由于下面OVER函数导致，换用其他方式改写语句就不会了。
SELECT  
...,
IF(tmp.personQuantity = 0, 0, ROUND(tmp.personQuantity / SUM(tmp.personQuantity) OVER (), 4)) AS ratio,
...
FROM (
 SELECT ...,
 BITMAP_COUNT(BITMAP_AND(b.user_id, fil.user_id)) AS personQuantity
 FROM ...
) AS tmp
ORDER BY personQuantity DESC
看官方文档是支持OVER的，但是不知道为什么这样使用会导致集群崩溃。
